### PR TITLE
feat: integrate SSO to i.ai

### PR DESCRIPTION
**Pull request by nboyse**
## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,7 +162,7 @@ make tf_init env=<ENVIRONMENT>
 make tf_apply env=<ENVIRONMENT>
 ```
 
-where ENVIRONMENT is one of `dev`, `preprod` or `prod`
+where ENVIRONMENT is one of `dev`, `sandbox`, `preprod` or `prod`
```
Comment (252afh) [inline README.md:165]: Do you have 4 envs? We have dev, preprod and prod, where preprod is just another word for sandbox. If this is coming into our repo, we'd rather keep preprod and ditch sandbox

File: django_app/redbox_app/redbox_core/models.py [hunk 2]
```diff
@@ -104,6 +106,38 @@ def __str__(self) -> str:
...
 
+class SSOUserManager(BaseSSOUserManager):
+
+    use_in_migrations = True
+
+    def _create_user(self, username, password, **extra_fields):
+        """Create and save a User with the given email and password."""
+        if not username:
+            raise ValueError("The given email must be set")
+        # email = self.normalize_email(email)
```
Comment (252afh) [inline django_app/redbox_app/redbox_core/models.py:117]: Remove comment

File: django_app/redbox_app/redbox_core/models.py [hunk 3]
```diff
@@ -104,6 +106,38 @@ def __str__(self) -> str:
...
+        extra_fields.setdefault("is_superuser", False)
+        return self._create_user(username, password, **extra_fields)
+
+    def create_superuser(self, username, password=None, **extra_fields):
+        """Create and save a SuperUser with the given email and password."""
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+
+        if extra_fields.get("is_staff") is not True:
+            raise ValueError("Superuser must have is_staff=True.")
```
Comment (252afh) [inline django_app/redbox_app/redbox_core/models.py:135]: Be consistent with punctuation. The error at ln116 has no fullstop, these do

File: django_app/redbox_app/setting_enums.py [hunk 4]
```diff
@@ -25,6 +25,8 @@ def uses_minio(self) -> bool:
     DEV = ("DEV", False, ["redbox-dev.ai.cabinetoffice.gov.uk", *ADDITIONAL_HOSTS])
     PREPROD = ("PREPROD", False, ["redbox-preprod.ai.cabinetoffice.gov.uk", *ADDITIONAL_HOSTS])
     PROD = ("PROD", False, ["redbox.ai.cabinetoffice.gov.uk", *ADDITIONAL_HOSTS])
+    DEVELOPMENT = ("DEVELOPMENT", False, [*ADDITIONAL_HOSTS])
+    SANDBOX = ("SANDBOX", False, [*ADDITIONAL_HOSTS])
```
Comment (252afh) [inline django_app/redbox_app/setting_enums.py:29]: We shouldn't be adding an extra env that we don't have

File: pyproject.toml [hunk 5]
```diff
@@ -13,6 +13,7 @@ readme = "README.md"
 [tool.poetry.dependencies]
 python = ">=3.12,<3.13"
 
+django-staff-sso-client = "^4.3.0"
```
Comment (252afh) [inline pyproject.toml:16]: I'm not a big fan of using internal packages for something that we have django-socials for, but we're already unique in the login methods for this project that it won't make too much difference

Comment (gecBurton) [reply pyproject.toml:16]: @nboyse is this DBT specific, or okta? who else could use this (in principle)?

**Review by gecBurton — APPROVED** [review]
LGTM

**Pull request merged**